libkbfs/conflict_resolver.go

```diff
@@ -1693,22 +1691,79 @@ func collapseActions(unmergedChains *crChains,
 			continue
 		}
 
+		// If this is a directory with setAttr(mtime)-related actions,
+		// just those action should be collapsed into the parent.
+		if !chain.isFile() {
+			var parentActions crActionList
+			var otherDirActions crActionList
+			for _, action := range fileActions {
+				moved := false
+				switch realAction := action.(type) {
+				case *copyUnmergedAttrAction:
+					if realAction.attr[0] == mtimeAttr && !realAction.moved {
+						realAction.moved = true
+						parentActions = append(parentActions, realAction)
+						moved = true
+					}
+				case *renameUnmergedAction:
+					if realAction.causedByAttr == mtimeAttr &&
+						!realAction.moved {
+						realAction.moved = true
+						parentActions = append(parentActions, realAction)
+						moved = true
+					}
+				}
+				if !moved {
```

Comment:
Because of the `if` statements in the above cases.  Type switches don't support fallthrough as far as I know.